subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -121,10 +121,33 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
 				       buf, sizeof(buf))));
 		}
 
-		if_addr = net_if_ipv6_addr_add(
+		/* Thread and SLAAC are clearly AUTOCONF, handle
+		 * manual/NCP addresses in the same way
+		 */
+		if ((address->mAddressOrigin == OT_ADDRESS_ORIGIN_THREAD) ||
+		    (address->mAddressOrigin == OT_ADDRESS_ORIGIN_SLAAC)) {
+			if_addr = net_if_ipv6_addr_add(
 					context->iface,
 					(struct in6_addr *)(&address->mAddress),
 					NET_ADDR_AUTOCONF, 0);
+		} else if (
```

Comment:
nit: I'd break the line instead of moving it all below, IMO currently it blends with the code and is less readable.
```
		} else if (address->mAddressOrigin == 
			   OT_ADDRESS_ORIGIN_DHCPV6) {
```

Reply:
Addressed with the latest force-push